metrics/api/v1/historical_handlers.go

```diff
@@ -0,0 +1,770 @@
+	podContainerAggregations(request *restful.Request, response *restful.Response)
+	freeContainerAggregations(request *restful.Request, response *restful.Response)
+	podListAggregations(request *restful.Request, response *restful.Response)
+
+	isRunningInKubernetes() bool
+}
+
+// addAggregationRoutes adds routes to a webservice which point to a metricsAggregationFetcher's methods
+func addAggregationRoutes(a metricsAggregationFetcher, ws *restful.WebService) {
+	// The /metrics-aggregated/{aggregations}/{metric-name} endpoint exposes some aggregations for the Cluster entity of the historical API.
+	ws.Route(ws.GET("/metrics-aggregated/{aggregations}/{metric-name:*}").
+		To(metrics.InstrumentRouteFunc("clusterMetrics", a.clusterAggregations)).
+		Doc("Export some cluster-level metric aggregations").
+		Operation("clusterAggregations").
+		Param(ws.PathParameter("aggregations", "A comma-separated list of requested aggregations").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metric").DataType("string")).
+		Param(ws.QueryParameter("end", "End time for requested metric").DataType("string")).
+		Writes(types.MetricAggregationResult{}))
+
+	// The /nodes/{node-name}/metrics-aggregated/{aggregations}/{metric-name} endpoint exposes some aggregations for a Node entity of the historical API.
+	// The {node-name} parameter is the hostname of a specific node.
+	ws.Route(ws.GET("/nodes/{node-name}/metrics-aggregated/{aggregations}/{metric-name:*}").
+		To(metrics.InstrumentRouteFunc("nodeMetrics", a.nodeAggregations)).
+		Doc("Export a node-level metric").
+		Operation("nodeAggregations").
+		Param(ws.PathParameter("node-name", "The name of the node to lookup").DataType("string")).
+		Param(ws.PathParameter("aggregations", "A comma-separated list of requested aggregations").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metric").DataType("string")).
+		Param(ws.QueryParameter("end", "End time for requested metric").DataType("string")).
+		Writes(types.MetricAggregationResult{}))
+
+	if a.isRunningInKubernetes() {
+		// The /namespaces/{namespace-name}/metrics-aggregated/{aggregations}/{metric-name} endpoint exposes some aggregations
+		// for a Namespace entity of the historical API.
+		ws.Route(ws.GET("/namespaces/{namespace-name}/metrics-aggregated/{aggregations}/{metric-name:*}").
+			To(metrics.InstrumentRouteFunc("namespaceMetrics", a.namespaceAggregations)).
+			Doc("Export some namespace-level metric aggregations").
+			Operation("namespaceAggregations").
+			Param(ws.PathParameter("namespace-name", "The name of the namespace to lookup").DataType("string")).
+			Param(ws.PathParameter("aggregations", "A comma-separated list of requested aggregations").DataType("string")).
```

Comment:
The aggregations documented in the proposal.  Do you want that documented elsewhere, or a subpath which just returns that list?